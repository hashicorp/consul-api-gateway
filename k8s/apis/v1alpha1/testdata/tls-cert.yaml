apiVersion: polar.hashicorp.com/v1alpha1
kind: GatewayClassConfig
metadata:
  name: test-gateway-class-config
spec:
  consul:
    scheme: https
    caSecret: super-secret
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: test-gateway-class
spec:
  controller: "hashicorp.com/polar-gateway-controller"
  parametersRef:
    group: polar.hashicorp.com
    kind: GatewayClassConfig
    name: test-gateway-class-config
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: tls-cert-test
spec:
  gatewayClassName: test-gateway-class
  listeners:  
  - protocol: HTTP
    port: 80
    name: http
    allowedRoutes:
      namespaces:
        from: Same