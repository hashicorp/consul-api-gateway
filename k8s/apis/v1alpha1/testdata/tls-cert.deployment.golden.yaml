metadata:
  creationTimestamp: null
  labels:
    managedBy: polar
    name: polar-tls-cert-test
  name: tls-cert-test
spec:
  selector:
    matchLabels:
      managedBy: polar
      name: polar-tls-cert-test
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        managedBy: polar
        name: polar-tls-cert-test
    spec:
      containers:
      - command:
        - /bootstrap/polar
        - exec
        - -log-json
        - -log-level
        - info
        - -gateway-host
        - $(IP)
        - -gateway-name
        - tls-cert-test
        - -consul-http-address
        - $(HOST_IP)
        - -consul-http-port
        - "8500"
        - -consul-xds-port
        - "8502"
        - -envoy-bootstrap-path
        - /bootstrap/envoy.json
        - -envoy-sds-address
        - polar-controller.default.svc.cluster.local
        - -envoy-sds-port
        - "9090"
        - -consul-ca-cert-file
        - /ca/tls.crt
        env:
        - name: IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: HOST_IP
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
        image: envoyproxy/envoy:v1.19-latest
        name: polar
        ports:
        - containerPort: 20000
          name: ready
          protocol: TCP
        - containerPort: 80
          name: http
          protocol: TCP
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /ca
          name: ca
          readOnly: true
      initContainers:
      - command:
        - cp
        - /bin/polar
        - /bootstrap/polar
        image: hashicorp/polar:0.1.0
        name: polar-init
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /ca
          name: ca
          readOnly: true
      volumes:
      - emptyDir: {}
        name: bootstrap
      - emptyDir: {}
        name: certs
      - name: ca
        secret:
          secretName: super-secret
status: {}
